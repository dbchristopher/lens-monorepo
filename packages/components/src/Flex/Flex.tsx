import React, { FunctionComponent, Ref } from 'react';
import styled, { css, StyledComponent } from 'styled-components';
import { BoxBasePropsWithout, BoxFlexProps } from '../Box/Box';
import Box from '../Box';

/**
 * styled-system has its own FlexProps, so we call this one FlexBoxProps to disambiguate.
 */
export interface FlexProps
  extends BoxBasePropsWithout<HTMLDivElement, 'display'>,
    BoxFlexProps {
  hidden?: boolean;
}

type ComponentType = FunctionComponent<FlexProps>;
type StyledComponentType = StyledComponent<ComponentType, FlexProps>;

const InternalFlex: ComponentType = props => {
  return (
    <Box display="flex" {...props}>
      {props.children}
    </Box>
  );
};

const FlexFactory = React.forwardRef<StyledComponentType, FlexProps>(
  (props: FlexProps, ref: Ref<StyledComponentType>) => (
    <InternalFlex ref={ref} {...props} />
  )
);

const Flex = styled<ComponentType>(FlexFactory)`
  ${(props: FlexProps) => hidden(props.hidden)};
`;

const hidden = (hide?: boolean) =>
  hide
    ? css`
        display: none;
      `
    : false;

export default Flex;
