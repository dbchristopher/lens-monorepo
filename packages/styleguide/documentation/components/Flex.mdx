---
name: Flex
menu: Layout
---

import { Playground } from 'docz';
import { Flex, FlexItem, Box, Text /* Icon */ } from 'looker-lens-components';

# Flex

The `<Flex />` component creates a flex container for you to place `<FlexItem />` components inside. It lets you utilize [flex box](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout) to create a wide variety of layouts, with powerful alignment, directional, and space distribution abilities.

If you are not familiar with flex box, [here is a great guide](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) for how it works.

## Basic Usage

The `<Flex />` component has the ability to control the layout direction and order of its children. By default the children will layout in a row.

<Playground>
  <Flex>
    <FlexItem>👋</FlexItem>
    <FlexItem>💪</FlexItem>
    <FlexItem>📦</FlexItem>
  </Flex>
</Playground>

## A real world example

Using our `<Flex>` and `<FlexItem>` components we can quickly scaffold out real world use cases like a navigation bar.

<Playground>
  <Flex
    bg="palette.charcoal800"
    px="small"
    height="50px"
    alignItems="center"
    justifyContent="space-between"
  >
    <FlexItem mr="medium" pt="xsmall">
      Hamburger Icon
      {/* <Icon name="Hamburger" color="palette.white" size={24} /> */}
    </FlexItem>
    <FlexItem>
      Looker Logo
      {/* <Icon name="LookerLogo" color="palette.white" size={60} /> */}
    </FlexItem>
    <FlexItem flex="1">
      <Flex justifyContent="center">
        <Box px="small">
          <Text fontSize="xsmall" variant="inverted">
            Browse
            {/*  <Icon name="ExpandMore" /> */}
          </Text>
        </Box>
        <Box px="small">
          <Text fontSize="xsmall" variant="inverted">
            Explore
            {/* <Icon name="ExpandMore" /> */}
          </Text>
        </Box>
        <Box px="small">
          <Text fontSize="xsmall" variant="inverted">
            Develop
            {/* <Icon name="ExpandMore" /> */}
          </Text>
        </Box>
        <Box px="small">
          <Text fontSize="xsmall" variant="inverted">
            Admin
            {/* <Icon name="ExpandMore" /> */}
          </Text>
        </Box>
      </Flex>
    </FlexItem>
    <FlexItem>
      <Flex>
        <FlexItem mr="small">
          Search
          {/* <Icon name="Search" color="palette.white" size={24} /> */}
        </FlexItem>
        <FlexItem mr="small">
          Chat
          {/* <Icon name="Chat" color="palette.white" size={24} /> */}
        </FlexItem>
        <FlexItem>
          Group
          {/* <Icon name="Group" color="palette.white" size={24} /> */}
        </FlexItem>
      </Flex>
    </FlexItem>
  </Flex>
</Playground>

## Changing direction and order

You can change the layout direction and order of flex children by using the `flexDirection` property. Below we use the same children in each example but can adjust the layout quickly using `flexDirection`.

<Playground>
  <Flex>
    <Flex mr="large">
      <FlexItem>1️⃣</FlexItem>
      <FlexItem>2️⃣</FlexItem>
      <FlexItem>3️⃣</FlexItem>
    </Flex>

    <Flex flexDirection="column" mr="large">
      <FlexItem>1️⃣</FlexItem>
      <FlexItem>2️⃣</FlexItem>
      <FlexItem>3️⃣</FlexItem>
    </Flex>

    <Flex flexDirection="column-reverse" mr="large">
      <FlexItem>1️⃣</FlexItem>
      <FlexItem>2️⃣</FlexItem>
      <FlexItem>3️⃣</FlexItem>
    </Flex>

    <Flex flexDirection="row-reverse" mr="large">
      <FlexItem>1️⃣</FlexItem>
      <FlexItem>2️⃣</FlexItem>
      <FlexItem>3️⃣</FlexItem>
    </Flex>

  </Flex>
</Playground>

## Item distribution

The `<Flex>` component allows you to control how its children are distributed within its container. Using the `justifyContent` property you can adjust the distribution of the children. Below is a table with all the justify options:

```js noeditor
import { Flex } from './Flex';
import { FlexItem } from '../FlexItem';
import {
  Table,
  TableHead,
  TableRow,
  TableHeaderCell,
  TableBody,
  TableDataCell,
} from '../Table';
import { Code, Text } from '../Text';
<Table>
  <TableHead>
    <TableRow>
      <TableHeaderCell width="40%">Justify Property</TableHeaderCell>
      <TableHeaderCell>Behavior</TableHeaderCell>
    </TableRow>
  </TableHead>
  <TableBody>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>center</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">Pack items around the center</Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>flex-start</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">Pack items from the start</Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>flex-end</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">Pack items from the end</Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>space-between</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          Distribute items evenly. The first item is flush with the start, the
          last is flush with the end{' '}
        </Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>space-around</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          Distribute items evenly. Items have a half-size space on either end
        </Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>space-evenly</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          Distribute items evenly. Items have equal space around them
        </Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>stretch</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          Distribute items evenly. Items strechted to fit the container
        </Text>
      </TableDataCell>
    </TableRow>
  </TableBody>
</Table>;
```

Here is a visual example of how using `justifyContent` affects the distribution of items in a `<Flex />` component.

<Playground>
  <Flex justifyContent="flex-start" mb="medium" bg="palette.charcoal000">
    <Box>
      <Text fontSize="xxxxlarge">🕺</Text>
    </Box>
    <Box>
      <Text fontSize="xxxxlarge">🎶</Text>
    </Box>
    <Box>
      <Text fontSize="xxxxlarge">💃</Text>
    </Box>
  </Flex>

<Flex justifyContent="space-between" mb="medium" bg="palette.charcoal000">
  <Box>
    <Text fontSize="xxxxlarge">🕺</Text>
  </Box>
  <Box>
    <Text fontSize="xxxxlarge">🎶</Text>
  </Box>
  <Box>
    <Text fontSize="xxxxlarge">💃</Text>
  </Box>
</Flex>

<Flex justifyContent="space-around" mb="medium" bg="palette.charcoal000">
  <Box>
    <Text fontSize="xxxxlarge">🕺</Text>
  </Box>
  <Box>
    <Text fontSize="xxxxlarge">🎶</Text>
  </Box>
  <Box>
    <Text fontSize="xxxxlarge">💃</Text>
  </Box>
</Flex>

<Flex justifyContent="space-evenly" mb="medium" bg="palette.charcoal000">
  <Box>
    <Text fontSize="xxxxlarge">🕺</Text>
  </Box>
  <Box>
    <Text fontSize="xxxxlarge">🎶</Text>
  </Box>
  <Box>
    <Text fontSize="xxxxlarge">💃</Text>
  </Box>
</Flex>

<Flex justifyContent="center" mb="medium" bg="palette.charcoal000">
  <Box>
    <Text fontSize="xxxxlarge">🕺</Text>
  </Box>
  <Box>
    <Text fontSize="xxxxlarge">🎶</Text>
  </Box>
  <Box>
    <Text fontSize="xxxxlarge">💃</Text>
  </Box>
</Flex>

  <Flex justifyContent="flex-end" mb="medium" bg="palette.charcoal000">
    <Box>
      <Text fontSize="xxxxlarge">🕺</Text>
    </Box>
    <Box>
      <Text fontSize="xxxxlarge">🎶</Text>
    </Box>
    <Box>
      <Text fontSize="xxxxlarge">💃</Text>
    </Box>
  </Flex>
</Playground>

## Aligning Items

Flex box allows for quickly adjusting the alignment of items inside the flex container. Below are available alignment types:

```js noeditor
import {
  Table,
  TableHead,
  TableRow,
  TableHeaderCell,
  TableBody,
  TableDataCell,
} from '../Table';
import { Code, Text } from '../Text';
<Table>
  <TableHead>
    <TableRow>
      <TableHeaderCell width="40%">AlignItems Property</TableHeaderCell>
      <TableHeaderCell>Behavior</TableHeaderCell>
    </TableRow>
  </TableHead>
  <TableBody>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>center</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">Items are aligned center of container</Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>flex-start</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          Items are aligned to the start of the container
        </Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>flex-end</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          Items are aligned to the start of the container
        </Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>baseline</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          All items are aligned so their baselines align
        </Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>stretch</Code>
        </span>
        <Text fontSize="xsmall">(default)</Text>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">Items strechted to fit the container</Text>
      </TableDataCell>
    </TableRow>
  </TableBody>
</Table>;
```

Here is an example of using `alignItems` to adjust how items are aligned within the flex container.

<Playground>
  <Flex mb="xxlarge">
    <FlexItem p="xxlarge" bg="palette.purple200">
      Default (stretch)
    </FlexItem>
    <FlexItem p="large" bg="palette.blue200">
      One
    </FlexItem>
    <FlexItem p="medium" bg="palette.green200">
      Two
    </FlexItem>
    <FlexItem p="small" bg="palette.red200">
      Three
    </FlexItem>
  </Flex>

<Flex alignItems="center" mb="xxlarge">
  <FlexItem p="xxlarge" bg="palette.purple200">
    Center
  </FlexItem>
  <FlexItem p="large" bg="palette.blue200">
    One
  </FlexItem>
  <FlexItem p="medium" bg="palette.green200">
    Two
  </FlexItem>
  <FlexItem p="small" bg="palette.red200">
    Three
  </FlexItem>
</Flex>

<Flex alignItems="flex-start" mb="xxlarge">
  <FlexItem p="xxlarge" bg="palette.purple200">
    Start
  </FlexItem>
  <FlexItem p="large" bg="palette.blue200">
    Two
  </FlexItem>
  <FlexItem p="medium" bg="palette.green200">
    Three
  </FlexItem>
  <FlexItem p="small" bg="palette.red200">
    Four
  </FlexItem>
</Flex>

<Flex alignItems="flex-end" mb="xxlarge">
  <FlexItem p="xxlarge" bg="palette.purple200">
    End
  </FlexItem>
  <FlexItem p="large" bg="palette.blue200">
    Two
  </FlexItem>
  <FlexItem p="medium" bg="palette.green200">
    Three
  </FlexItem>
  <FlexItem p="small" bg="palette.red200">
    Four
  </FlexItem>
</Flex>

  <Flex alignItems="baseline" mb="xxlarge">
    <FlexItem p="xxlarge" bg="palette.purple200">
      Baseline
    </FlexItem>
    <FlexItem p="large" bg="palette.blue200">
      Two
    </FlexItem>
    <FlexItem p="medium" bg="palette.green200">
      Three
    </FlexItem>
    <FlexItem p="small" bg="palette.red200">
      Four
    </FlexItem>
  </Flex>
</Playground>

## Aligning Content

If you have multiple rows of flex items, the `alignContent` property determines how the rows are distributed within the flex container.

```js noeditor
import { Code, Text } from '../Text';
import {
  Table,
  TableHead,
  TableRow,
  TableHeaderCell,
  TableBody,
  TableDataCell,
} from '../Table';
<Table>
  <TableHead>
    <TableRow>
      <TableHeaderCell width="40%">AlignContent Property</TableHeaderCell>
      <TableHeaderCell>Behavior</TableHeaderCell>
    </TableRow>
  </TableHead>
  <TableBody>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>center</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          Lines are packed to the center of container
        </Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>flex-start</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          Lines are packed to the start of the container
        </Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>flex-end</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          Lines are packed to the start of the container
        </Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>space-between</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          Lines evenly distributed. First line at start of container, last line
          at end of container
        </Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>space-around</Code>
        </span>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">
          Lines evenly distributed. Even space around each line
        </Text>
      </TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>
        <span className="prop-code">
          <Code>stretch</Code>
        </span>{' '}
        <Text fontSize="xsmall">(default)</Text>
      </TableDataCell>
      <TableDataCell>
        <Text fontSize="xsmall">Items strechted to fit the container</Text>
      </TableDataCell>
    </TableRow>
  </TableBody>
</Table>;
```

<Playground>
<Flex justifyContent="space-between">
  <Flex
    alignContent="space-around"
    width="30%"
    height="150px"
    flexWrap="wrap"
    bg="palette.purple400"
  >
    <FlexItem width="70%" m="small" p="xsmall" bg="palette.purple200">
      Aligned w/
    </FlexItem>
    <FlexItem width="30%" m="small" p="xsmall" bg="palette.purple200">
      Space
    </FlexItem>
    <FlexItem width="50%" m="small" p="xsmall" bg="palette.purple200">
      Betwen
    </FlexItem>
  </Flex>

<Flex
  alignContent="flex-end"
  width="30%"
  height="150px"
  flexWrap="wrap"
  bg="palette.blue400"
>
  <FlexItem width="70%" m="small" p="xsmall" bg="palette.blue200">
    Aligned w/
  </FlexItem>
  <FlexItem width="30%" m="small" p="xsmall" bg="palette.blue200">
    Flex
  </FlexItem>
  <FlexItem width="50%" m="small" p="xsmall" bg="palette.blue200">
    End
  </FlexItem>
</Flex>

  <Flex
    alignContent="center"
    width="30%"
    height="150px"
    flexWrap="wrap"
    bg="palette.green400"
  >
    <FlexItem width="70%" m="small" p="xsmall" bg="palette.green200">
      Aligned w/
    </FlexItem>
    <FlexItem width="30%" m="small" p="xsmall" bg="palette.green200">
      Flex
    </FlexItem>
    <FlexItem width="50%" m="small" p="xsmall" bg="palette.green200">
      Center
    </FlexItem>
  </Flex>
</Flex>
</Playground>

## Wrapping Flex Items

The `flexWrap` property determines if flex items should be forced into a single line or if they can wrap onto multiple lines.

<Playground>
  <Flex width="80%" flexWrap="nowrap" mb="large">
    <FlexItem width="40%" p="large" bg="palette.red200">
      These Lines
    </FlexItem>
    <FlexItem width="40%" p="large" bg="palette.blue200">
      Will NOT
    </FlexItem>
    <FlexItem width="40%" p="large" bg="palette.green200">
      Wrap
    </FlexItem>
  </Flex>
  <Flex width="50%" flexWrap="wrap" mb="large">
    <FlexItem width="40%" p="large" bg="palette.red200">
      These Lines
    </FlexItem>
    <FlexItem width="40%" p="large" bg="palette.blue200">
      Will
    </FlexItem>
    <FlexItem width="40%" p="large" bg="palette.green200">
      Wrap
    </FlexItem>
  </Flex>
  <Flex width="50%" flexWrap="wrap-reverse" mb="large">
    <FlexItem width="40%" p="large" bg="palette.red200">
      These Lines
    </FlexItem>
    <FlexItem width="40%" p="large" bg="palette.blue200">
      Will
    </FlexItem>
    <FlexItem width="40%" p="large" bg="palette.green200">
      Wrap Reverse
    </FlexItem>
  </Flex>
</Playground>

## Responsive Behavior

Many of the properties you can set on the `<Flex />` component will accept an array of values that will then be applied at different breakpoints. There are 5 [breakpoints](/#!/Breakpoints) slots that you can target. Here is an example of what that looks like.

<Playground>
  <Flex
    flexDirection={[
      'column', // column up to the first breakpoint
      null, // stay a column past first breakpoint
      'row', // switch to row layout after second breakpoint
    ]}
  >
    <FlexItem p="medium" bg="palette.yellow200">
      👋
    </FlexItem>
    <FlexItem p="medium" bg="palette.green200">
      💪
    </FlexItem>
    <FlexItem p="medium" bg="palette.purple200">
      📦
    </FlexItem>
  </Flex>
</Playground>
