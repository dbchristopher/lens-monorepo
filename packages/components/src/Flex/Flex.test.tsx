import 'jest-styled-components'
import * as React from 'react'
import { assertSnapshot } from '../../../test/utils/snapshot'
import { Flex } from './Flex'

test('Flex default ', () => {
  assertSnapshot(
    <Flex>
      <div>🥑</div>
      <div>🐛</div>
    </Flex>
  )
})

test('Flex supports justifyContent ', () => {
  assertSnapshot(<Flex justifyContent="center" />)
})

test('Flex supports alignItems ', () => {
  assertSnapshot(<Flex alignItems="center" />)
})

test('Flex supports alignContent ', () => {
  assertSnapshot(<Flex alignContent="start" />)
})

test('Flex supports flexDirection ', () => {
  assertSnapshot(<Flex flexDirection="row-reverse" />)
})

test('Flex supports flexWrap ', () => {
  assertSnapshot(<Flex flexWrap="nowrap" />)
})

test('Flex can be hidden ', () => {
  assertSnapshot(<Flex hidden />)
})
