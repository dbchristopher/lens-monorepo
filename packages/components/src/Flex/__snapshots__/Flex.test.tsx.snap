// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex can be hidden  1`] = `
.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: none;
}

<div
  className="c0 c1"
  hidden={true}
/>
`;

exports[`Flex default  1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
>
  <div>
    ü•ë
  </div>
  <div>
    üêõ
  </div>
</div>
`;

exports[`Flex supports alignContent  1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
/>
`;

exports[`Flex supports alignItems  1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
/>
`;

exports[`Flex supports flexDirection  1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

<div
  className="c0"
/>
`;

exports[`Flex supports flexWrap  1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

<div
  className="c0"
/>
`;

exports[`Flex supports justifyContent  1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;
